### ISSUE TITLE
Upgrade Vue to 3.0

### ISSUE THREAD

**Issue by agriffard** [OrchardCMS/OrchardCore#7110]
Migrate the ResourceManifest, the dependent scripts, the ContentPicker and the LocalizationContentPicker to use vue.js 3.0

https://www.npmjs.com/package/vue/v/3.0.0

/cc @Skrypt

**Comment by jptissot** [comment]
https://github.com/SortableJS/Vue.Draggable/issues/881

**Comment by Skrypt** [comment]
Major version = breaking changes.

**Comment by jptissot** [comment]
We should wait a couple months for the ecosystem to migrate to it and fix  the bugs.

**Comment by Skrypt** [comment]
Or at least if we migrate do manual testing to make sure nothing is breaking. I'm pretty sure it will break all the draggable components.

**Comment by agriffard** [comment]
> [SortableJS/Vue.Draggable#881](https://github.com/SortableJS/Vue.Draggable/issues/881)

There is a https://www.npmjs.com/package/vue-draggable-next package available.

Not sure if we should switch to this one to be compatible with vue 3?

**Comment by jptissot** [comment]
I would wait a couple months for others to test the vue components for us 😁

**Comment by Skrypt** [comment]
Well create the PR guys if you think it could potentially work. But test a lot before merging.

**Comment by Piedone** [comment]
@agriffard any specific reason you closed this? Eventually, we should upgrade. And since then, Vue.Draggable
 is also upgraded: https://github.com/SortableJS/Vue.Draggable/issues/942

**Comment by agriffard** [comment]
There is no CDN available for the latest version:
https://www.npmjs.com/package/vuedraggable/v/4.1.0

The url they give returns a 404:
https://github.com/SortableJS/vue.draggable.next#with-direct-link

`<script src="//cdnjs.cloudflare.com/ajax/libs/Vue.Draggable/4.0.0/vuedraggable.umd.min.js"></script>`

**Comment by Piedone** [comment]
That's a bug in Vue.Draggable then that we need before this can be addressed. However, any reason why not keep this issue open?

**Comment by Piedone** [comment]
Hmm, there's https://github.com/SortableJS/Vue.Draggable and https://github.com/SortableJS/vue.draggable.next. The latter is the project going forward with Vue v3. It appears to have restarted versioning (though this is inconsistent on GitHub/NPM vs the CDN) and the new CDN is here: https://unpkg.com/vue-draggable-next@2.1.1/dist/vue-draggable-next.global.js See: https://github.com/SortableJS/vue.draggable.next/issues/64

**Comment by Piedone** [comment]
We need this ASAP now, since Vue 2 reaches its EOL by the end of the year: https://v2.vuejs.org/lts/.

**Comment by Skrypt** [comment]
You have some WIP there: https://github.com/OrchardCMS/OrchardCore/pull/14256

**Comment by Piedone** [comment]
Awesome! Do you need some help with that?

**Comment by Skrypt** [comment]
Yeah, any help would be appreciated. Basically my PR adds the ability to keep Vue 2 and Vue 3 in the solution ; so that we can split the tasks in different PR's afterward. Though, I still need to migrate the `media library picker` component to be able to say that the PR is a complete Media Library upgrade.

Let me know if you need more details we can do a Teams call.

**Comment by Skrypt** [comment]
Another suggestion would be to convert all of these to Blazor components. Get rid of the Vue dependency and find something else than the JQuery UI upload component to do this. I upgraded the Vue component to use it as a standalone app for a different project but I definitely would see this rewritten as a Blazor component. @sebastienros

**Comment by Piedone** [comment]
I'd upgrade Vue instead of a more drastic change, at this point at least. Also, the Vue dependency itself needs to be upgraded in any case, so module authors can depend on the new version.

**Comment by Skrypt** [comment]
The way the OC admin UI is built is to not depend on Vue at all. Just Bootstrap and some custom Javascript.
It is possible to keep Vue 2 and Vue 3 in the resourceManager for the time that people will migrate their custom modules to Vue 3.

I think most of the work is about migrating the Media Library and the CORS module UI. Of course now that we have a good migration of the Media Library here I'd say just push to migrate everything else for now. We could alternatively create a small Blazor component for the OptionEditor ... we have the choice as this is a smaller component. But definitely agree that the Media Library would take way more time to Blazorize so nothing planned in the short term for this.

> Vue dependency itself needs to be upgraded in any case

I think it needs to be added in the resourceManager along with the Vue 2 dependency instead.

**Comment by Skrypt** [comment]
Also wanted to have a clear opinion about Blazor components from @sebastienros first. That's why I pinged him.

**Comment by Piedone** [comment]
Where was this done @agriffard? https://github.com/OrchardCMS/OrchardCore/pull/14256 is open.

**Comment by agriffard** [comment]
Closed in favor of https://github.com/OrchardCMS/OrchardCore/pull/14256

**Comment by Piedone** [comment]
But that's a PR, we still need an issue to keep track of the request.

**Comment by Skrypt** [comment]
This one is done. The only thing remaining is to refactor some components in OC.
